paddle/fluid/pybind/tensor_py.h

```diff
@@ -241,8 +245,8 @@ void SetTensorFromPyArray(framework::Tensor *self, const py::object &obj,
   } else if (py::isinstance<py::array_t<uint16_t>>(array)) {
     // TODO(cql): temporary keeping uint16, which is used for casting float16
     // before. It should be depracated later.
-    SetTensorFromPyArrayT<paddle::platform::float16, P>(self, array, place,
-                                                        zero_copy);
+    SetTensorFromPyArrayT<paddle::platform::bfloat16, P>(self, array, place,
+                                                         zero_copy);
```

Comment:
Can we safely replace that? Isn't it used somewhere in GPU operators to convert `uint16_t` numpy arrays to `float16` ones?

Reply:
I checked that since NumPy supports type float16, tests have been changed.
@zhiqiu  Do you know if conversion from uint16 to float16 is still needed?

Reply:
Since PR #19964, ` Tensor.set()` method supports  numpy array of `float16` directly, which means conversion from `uint16` to `float16` is not needed if user want to get a `float16` tensor.

Reply:
Thank you @zhiqiu. It's good news.